### DISCUSSION TITLE
iTwin Snapshot APP api

### DISCUSSION THREAD

**Discussion by namsancow** [iTwin/itwinjs-core#1192] [Q&A] [answered]
Is there any method to convert design files as *.bim file automatically ?

I am using iTwin Snapshot APP, It is not convenient to use this APP every time.

I also checked "SnapshotDb.createFrom", "SnapshotDb.createEmpty" in core/backend/src/IModelDb.ts.
However "SnapshotDb.createFrom" just copys one *.bim file to another *.bim file. It has no function of converting.
And "SnapshotDb.createEmpty" just makes empty *.bim file. 

Is there iTwin Snapshot APP api or auto-converting function in imodeljs?

Please any advice with experience for this matter.

**Comment by pmconne** [comment, marked as answer]
> Is there iTwin Snapshot APP api or auto-converting function in imodeljs?

No. What are you doing that requires running iTwin Snapshot so frequently?

**Reply by namsancow** [reply]
Many thanks for your clear answer.
Actually, I do not use iTwin Snapshot App frequently. 

As I know, imodel file converter have changed from iTwin synchronizer(old version) to iTwin Snapshot(new version).
I have used iTwin synchronizer from 2020. But one day, it suddenly didn't work without notice.
I found out the converter was changed as iTwin Snapshot after searching the contents in iTwin.js.

I guess that the Snapshot converter will change and upgrade continually.
And, I posted this question because I couldn't find a special explanation for the Snapshot converter in git's imodeljs.

**Reply by pmconne** [reply]
iTwin Synchronizer and iTwin Snapshot are two different applications for two different purposes. Synchronizer does incremental conversion to iModel retaining the change history. Snapshot does one-time conversion with no change history. Both continue to be supported and updated.
